test/trento/domain/sap_system/sap_system_test.exs

```diff
@@ -26,7 +26,138 @@ defmodule Trento.SapSystemTest do
   alias Trento.Domain.SapSystem
 
   describe "SAP System registration" do
-    test "should create an incomplete SAP system aggregate and register a database instance" do
+    test "should fail when a sap system not exists and the database instance has Secondary role with system replication ACTIVE" do
+      command =
+        build(:register_database_instance_command,
+          system_replication: "Secondary",
+          system_replication_status: "ACTIVE"
+        )
+
+      assert_error(
+        command,
+        {:error, :sap_system_not_registered}
+      )
+    end
+
+    test "should fail when a sap system already exists and the database instance has Secondary role with system replication ACTIVE" do
```

Comment:
I don't understand this scenario. Once we have a sap system with a primary, other secondary nodes should be added to the instances list

Reply:
yes, removed